bw2io/backup.py

```diff
@@ -55,40 +62,155 @@ def backup_data_directory(
     timestamp_str = (
         f'.{datetime.datetime.now().strftime("%d-%B-%Y-%I-%M%p")}' if timestamp else ""
     )
-    backup_filename = f"brightway2-data-backup{timestamp_str}.tar.gz"
+    backup_filename = f"brightway2-data-backup{file_suffix}{timestamp_str}.tar.gz"
     fp = dir_backup / backup_filename
 
     # Create the backup archive
-    print("Creating backup archive of data directory - this could take a few minutes...")
+    print(
+        "Creating backup archive of data directory - this could take a few minutes..."
+    )
     with tarfile.open(fp, "w:gz") as tar:
         data_directory = Path(projects._base_data_dir)
         tar.add(data_directory, arcname=data_directory.name)
 
     print(f"Saved to: {fp}")
-    
+
     return fp
 
 
+def restore_data_directory(
+    fp: Union[str, Path],
+    new_data_dir: Optional[Union[str, Path]] = None,
+):
+    """
+    Restores a brightway data directory from a tar.gz file. If the data directory already exists, you must confirm that you want to delete it.
+    If ``new_data_dir`` is specified, the data directory will be restored to that location. Otherwise, the data directory will be restored to the default location. e.g: ``~/.local/share/brightway3`` on Linux.
+
+    Parameters
+    ----------
+    fp (Union[str, Path]):
+        The file path of the tar.gz file.
+    new_data_dir (Optional[Union[str, Path]]):
+        The new data directory path.
+
+    Raises
+    ------
+    ValueError:
+        If the file doesn't exist
+    PermissionError:
+        If the data directory is not writable.
+    Exception:
+        If there's an attempted path traversal in the tar file.
+
+    Returns
+    -------
+    data_dir (Path):
+        The path of the restored data directory.
+
+    See Also
+    --------
+    bw2io.backup.backup_data_directory: To backup the data directory.
+    """
+
+    # check if file exists
+    fp = Path(fp)
+    if not fp.is_file():
+        raise ValueError(f"Can't find file at path: {fp}")
+
+    # if new_data_dir, set the environment variable (this will not be persistent, I only know how to do that on Linux or in a venv...)
+    if new_data_dir:
+        os.environ["BRIGHTWAY2_DIR"] = str(new_data_dir)
```

Comment:
I don't think this will work.

If I understand correctly, the idea is that after importing the data directory, you can use it from `bw2data`. But `bw2io` has a dependency on `bw2data`, which means it has read the environment variables already and chosen the base data directory. Setting the environment variable now won't change that.

Instead, I think you should add an input argument to this function: `use_data_dir` (bool) (or similar), which would call [`change_base_directories`](https://github.com/brightway-lca/brightway2-data/blob/main/bw2data/project.py#L158).